test/logger_test.rb

```diff
@@ -0,0 +1,79 @@
+require 'test_helper'
+
+describe Lotus::Logger do
+
+  before do
+    #clear defined class
+    Object.send(:remove_const, :TestLogger) if Object.constants.include?(:TestLogger)
+  end
+
+  it 'like std logger, default log level is info' do
+    class TestLogger < Lotus::Logger; end
+    TestLogger.new.info?.must_equal true
+  end
+
+  it 'always use STDOUT' do
+    output =
+      stub_stdout_constant do
+        class TestLogger < Lotus::Logger; end
+        logger = TestLogger.new
+        logger.info('foo')
+      end
+    output.must_match /foo/
+  end
+
+  it 'has app_tag when log' do
+    output =
+      stub_stdout_constant do
+        module App; class TestLogger < Lotus::Logger; end; end
+        logger = App::TestLogger.new
+        logger.info('foo')
+      end
+    output.must_match /App/
+  end
+
+  it 'raise exception if override logger without app_tag' do
+    class TestLogger < Lotus::Logger; def app_tag; end; end
+    exception = -> { TestLogger.new.info('anything') }.must_raise(RuntimeError)
+    exception.message.must_equal "you must set app_tag at Logger constructor like Lotus::Logger.new('foo')"
+  end
+
+  it 'has default app tag when not in any namespace' do
+    class TestLogger < Lotus::Logger; end
+    TestLogger.new.send(:app_tag).must_equal 'Shared'
+  end
+
+  it 'infer apptag from namespace' do
+    module App2
+      class TestLogger < Lotus::Logger;end
+      class Bar
+        def hoge
+          TestLogger.new.send(:app_tag).must_equal 'App2'
+        end
+      end
+    end
+    App2::Bar.new.hoge
+  end
+
+  it 'use custom app_tag from override class' do
+    class TestLogger < Lotus::Logger; def app_tag; 'bar'; end; end
+    output =
+      stub_stdout_constant do
+        TestLogger.new.info('')
+      end
+    output.must_match /bar/
+  end
+
+  it 'has default time format "%Y-%m-%dT%H:%M:%S.%6N "' do
+    format = "%Y-%m-%dT%H:%M:%S.%6N "
+    stub_time_now do
+      strtime = Time.now.strftime(format)
+      output =
+        stub_stdout_constant do
+          class TestLogger < Lotus::Logger;end
+          TestLogger.new.info('')
+        end
+      output.must_match strtime
```

Comment:
Because we have already stubbed the time in this spec, can you please do `output.must_equal expected`? It's useful in tests to see at least one complete logging output example.